Dockerfile

```diff
@@ -35,6 +35,21 @@ RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
       --ignore-src \
     && rm -rf /var/lib/apt/lists/*
 
+#default value of arg ELASTICA_ADD when not provided in the --build-arg
+ARG ELASTICA_ADD="false"
+
+# ros2-elastica dependencies
+RUN if [ "${ELASTICA_ADD}" = "false" ]; then\
+      echo 'Not adding Elastica';\
+    else\
+      echo "Adding Elastica " &&\
+      apt-get update &&\
+      apt-get install --no-install-recommends -y tzdata dirmngr gnupg2 build-essential python3-colcon-common-extensions python3-colcon-mixin python3-rosdep python3-vcstool python3-pip  povray python3-tk &&\
```

Comment:
Please move the installation of `tzdata  dirmngr gnupg2 dirmngr build-essential python3-colcon-common-extensions python3-colcon-mixin python3-rosdep python3-vcstool python3-pip  python3-tk` to line 8. I think the are quite general and should be installed in any case even without elastica

Reply:
Yeah those were in my Dockerfile. Let me shift these general dependencies with others.

Reply:
Did your Dockerfile build on `ros:foxy` as well?

Reply:
Yes

Reply:
I have removed all of them except the last three (as they were there in the base image  of ros:foxy)